uint32 lowHue
uint32 highHue
uint32 lowSaturation
uint32 highSaturation
uint32 lowValue
uint32 highValue
